tuxemon/states/multiplayer/__init__.py

```diff
@@ -0,0 +1,148 @@
+# SPDX-License-Identifier: GPL-3.0
+# Copyright (c) 2014-2023 William Edwards <shadowapex@gmail.com>, Benjamin Bean <superman2k5@gmail.com>
+from __future__ import annotations
+
+from collections.abc import Callable, Generator
+
+import pygame_menu
+
+from tuxemon.animation import Animation
+from tuxemon.locale import T
+from tuxemon.menu.input import InputMenu
+from tuxemon.menu.interface import MenuItem
+from tuxemon.menu.menu import PopUpMenu, PygameMenuState
+from tuxemon.session import local_session
+from tuxemon.tools import open_dialog
+
+MenuGameObj = Callable[[], object]
+
+
+def add_menu_items(
+    menu: pygame_menu.Menu,
+    items: list[tuple[str, MenuGameObj]],
+) -> None:
+    for key, callback in items:
+        label = T.translate(key).upper()
+        menu.add.button(label, callback)
+
+
+class MultiplayerMenu(PygameMenuState):
+    """MP Menu
+
+    code salvaged from commit 6fa20da714c7b794cbe1e8a22168fa66cda13a9e
+    """
+
+    shrink_to_items = True
+
+    def __init__(self) -> None:
+        super().__init__()
+
+        menu: list[tuple[str, MenuGameObj]] = []
```

Comment:
minor, but when you are making a list, you don't need to use `append` for each item if it doesn't change.
```python
menu: list[tuple[str, MenuGameObj]] = [
   ("multiplayer_host_game", self.host_game),
   ("multiplayer_scan_games", self.scan_for_games),
   ("multiplayer_join_game", self.join_by_ip),
]
```